src/python/foglamp/device/http_south_device.py

```diff
@@ -0,0 +1,188 @@
+# -*- coding: utf-8 -*-
+
+# FOGLAMP_BEGIN
+# See: http://foglamp.readthedocs.io/
+# FOGLAMP_END
+
+"""HTTP Listener handler for sensor readings"""
+import sys
+from aiohttp import web
+import asyncio
+from foglamp import logger
+from foglamp.web import middleware
+from foglamp.device.ingest import Ingest
+
+__author__ = "Amarendra K Sinha"
+__copyright__ = "Copyright (c) 2017 OSIsoft, LLC"
+__license__ = "Apache 2.0"
+__version__ = "${VERSION}"
+
+_LOGGER = logger.setup(__name__)
+
+_CONFIG_CATEGORY_NAME = 'http_south'
+_CONFIG_CATEGORY_DESCRIPTION = 'South Plugin HTTP Listener'
+_DEFAULT_CONFIG = {
+    'plugin': {
+         'description': 'http_south_device',
+         'type': 'string',
+         'default': 'http_south'
+    },
+    'port': {
+        'description': 'Port to listen on',
+        'type': 'integer',
+        'default': '6683',
+    },
+    'host': {
+        'description': 'URI to accept data on',
+        'type': 'string',
+        'default': '0.0.0.0',
+    },
+    'uri': {
+        'description': 'URI to accept data on',
+        'type': 'string',
+        'default': 'sensor-reading',
+    }
```

Comment:
I'm not sure, if we should have only 1 key for url as `http(s)://host:port/uri_handler` or url composing units; do we have the (scaling) capability here to be SSL enabled later or sooner? (and hence the protocol can also be httpS vs http)

Reply:
This calls for an architecture level decision and discussion and this issue can be taken up separately.

Reply:
This fits the same model as CoAP ands hence matches the requirement in the Jira.